abstract-factory/src/main/java/com/iluwatar/abstractfactory/App.java

```diff
@@ -41,84 +40,14 @@
  * and its implementations ( {@link ElfKingdomFactory}, {@link OrcKingdomFactory}). The example uses
  * both concrete implementations to create a king, a castle and an army.
  */
-public class App {
+public class App implements Runnable {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(App.class);
+  private static Logger log = LoggerFactory.getLogger(App.class);
 
-  private King king;
-  private Castle castle;
-  private Army army;
+  private final Kingdom kingdom = new Kingdom();
```

Comment:
You can refactor it to

```
Supplier<Kingdom> kingdomSupplier = Kingdom::new

...

public Kingdom getKingdom() {
    return kingdomSupplier.get();
}
```

but its upto you

Reply:
I don't quite understand why. Can you explain it in more detail?